src/internet_identity/src/state/flow_states.rs

```diff
@@ -0,0 +1,131 @@
+use crate::anchor_management::registration::Base64;
+use crate::MINUTE_NS;
+use candid::Principal;
+use ic_cdk::api::time;
+use internet_identity_interface::internet_identity::types::Timestamp;
+use std::collections::{HashMap, VecDeque};
+
+// Expiration for a registration flow, the same as the captcha expiry
+const FLOW_EXPIRATION_NS: u64 = 5 * MINUTE_NS;
+
+#[derive(Default)]
+pub struct FlowStates {
```

Comment:
@lmuntaner: Ultimately, I decided against mixing flow states with temp_keys. I feel this is cleaner in terms of separation.
But it means that the "expiration logic" here and temp_keys is duplicated. I don't think it is big enough to warrant extracting that, but up to you.

Reply:
So now the temp keys won't be used to track the registration flow, right?

Do we need the enum for the TempKeyId still?

Reply:
No, I will revert #2618. I'll do that after this is merged.